tests/storage-interface.rs

```diff
@@ -0,0 +1,113 @@
+use eeprom24x::{Eeprom24x, Error, Storage};
+use embedded_hal_mock::i2c::{Mock as I2cMock, Transaction as I2cTrans};
+use embedded_storage::{ReadStorage, Storage as _};
+mod common;
+use crate::common::{
+    destroy, new_24x00, new_24x01, new_24x02, new_24x04, new_24x08, new_24x128, new_24x16,
+    new_24x256, new_24x32, new_24x512, new_24x64, new_24xm01, new_24xm02, new_m24x01, new_m24x02,
+    DEV_ADDR,
+};
+
+struct MockCountDown;
+impl embedded_hal::timer::CountDown for MockCountDown {
+    type Time = core::time::Duration;
+    fn start<T>(&mut self, _count: T)
+    where
+        T: Into<core::time::Duration>,
+    {
+        // no-op, just mock
+    }
+    fn wait(&mut self) -> nb::Result<(), void::Void> {
```

Comment:
```suggestion
    fn wait(&mut self) -> nb::Result<(), ()> {
```
Could you just use `()` for the error type here and avoid the `void` dependency at all?